pallets/statistics/src/lib.rs

```diff
@@ -177,15 +178,14 @@ impl<T: Config> Module<T> {
     ) -> DispatchResult {
         // Check EA permissions for asset.
         let did = Self::ensure_asset_perms(origin, asset)?;
-
-        // Check StatType per Asset limit.
-        ensure!(
-            stat_types.len() < T::MaxStatsPerAsset::get() as usize,
-            Error::<T>::StatTypeLimitReached
-        );
+        // converting from a btreeset to a bounded version
+        let bounded_stat_types: BoundedBTreeSet<_, T::MaxStatsPerAsset> = stat_types
+            .clone()
+            .try_into()
+            .map_err(|_| Error::<T>::StatTypeLimitReached)?;
```

Comment:
```suggestion
        let stat_types: BoundedBTreeSet<_, T::MaxStatsPerAsset> = stat_types
            .try_into()
            .map_err(|_| Error::<T>::StatTypeLimitReached)?;
```
Same here just replace the old unbounded variable with the bounded type.

Reply:
Will make the changes